libsolidity/AST.cpp

```diff
@@ -110,6 +110,16 @@ void FunctionDefinition::checkTypeRequirements()
 	m_body->checkTypeRequirements();
 }
 
+std::string FunctionDefinition::getCanonicalSignature()
+{
+	auto parameters = getParameters();
+	std::string ret = getName() + "(";
```

Comment:
yeah good! I actually was just thinking about that. I wrote this function first and then noticed the existence of the FunctionType :)